### PR TITLE
Issue 15435

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

**Review by nollymar — COMMENTED** [review]
An IT should be added

**Review by nollymar — COMMENTED** [review]
Also, the signature of these methods shouldn't be altered

**Review by wezell — COMMENTED** [review]
@erickgonzalez any tests we need here?

File: dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java [hunk 1]
```diff
@@ -470,32 +459,30 @@ private void deleteInodeInDB(final String inode) throws DotDataException{
...
 
-            return new ContentletTransformer(dc.loadObjectResults()).asList();
+        final List<Map<String, Object>> results = dc.loadObjectResults();
+        final List<Contentlet> contentlets = new ArrayList<Contentlet>();
+
+        for(final Map<String,Object> map : results){
+            try {
+                contentlets.add(APILocator.getContentletAPI().find((String) map.get("inode"),APILocator.systemUser(),false));
+            } catch (DotSecurityException e) {
+                //Do nothing since never gonna throw DotSecurityException is using systemUser
```
Comment (oarrietadotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java]: maybe log the exception + the inode. Just to be safe.

File: dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java [hunk 2]
```diff
@@ -507,12 +494,25 @@ private void deleteInodeInDB(final String inode) throws DotDataException{
...
-        return new ContentletTransformer(dc.loadObjectResults()).asList();
+
+        final List<Map<String, Object>> results = dc.loadObjectResults();
+        final List<Contentlet> contentlets = new ArrayList<Contentlet>();
+
+        for(final Map<String,Object> map : results){
+            try {
+                contentlets.add(APILocator.getContentletAPI().find((String) map.get("inode"),APILocator.systemUser(),false));
+            } catch (DotSecurityException e) {
+                //Do nothing since never gonna throw DotSecurityException is using systemUser
```
Comment (oarrietadotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java]: maybe log the exception + the inode. Just to be safe.

**Review by nollymar — APPROVED** [review]

**Review by oarrietadotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**